c10/util/ThreadLocalDebugInfo.cpp

```diff
@@ -1,11 +1,11 @@
+#include <c10/util/ThreadLocal.h>
 #include <c10/util/ThreadLocalDebugInfo.h>
 
 namespace c10 {
 
-namespace {
 // NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
-thread_local std::shared_ptr<ThreadLocalDebugInfo> debug_info = nullptr;
-} // namespace
+C10_DEFINE_TLS_static(std::shared_ptr<ThreadLocalDebugInfo>, tls_debug_info);
```

Comment:
Could you add a `NOTE [ Android TLS handling]` in the comment in ThreadLocal.h and link it from here like https://github.com/pytorch/pytorch/blob/3d70ab08ae58b1cf12c09e12729c61d3e850f739/torch/csrc/autograd/variable.h#L98
Just to make sure we remove these not so nice macros as soon as this is solved.